File: lab/python/matematica_e_operadores/torre_hanoi_recursivo.py
def torre_hanoi(n, origem, destino, auxiliar):
    if n == 1:
        print(f"Mova disco de {origem} para {destino}")
    else:
        torre_hanoi(n - 1, origem, auxiliar, destino)
        print(f"Mova disco de {origem} para {destino}")
        torre_hanoi(n - 1, auxiliar, destino, origem)


# Exemplo: 3 discos da torre A para C, usando B como auxiliar
torre_hanoi(3, "A", "C", "B")

File: lab/javascript/matematica_e_operadores/torre_hanoi_recursivo.js
function torreHanoi(n, origem, destino, auxiliar) {
    if (n === 1) {
        console.log(`Mova disco de ${origem} para ${destino}`);
    } else {
        torreHanoi(n - 1, origem, auxiliar, destino);
        console.log(`Mova disco de ${origem} para ${destino}`);
        torreHanoi(n - 1, auxiliar, destino, origem);
    }
}

// Exemplo: 3 discos da torre A para C, usando B como auxiliar
torreHanoi(3, "A", "C", "B");
